cartographer/sensor/compressed_point_cloud.cc

```diff
@@ -145,9 +145,16 @@ CompressedPointCloud::CompressedPointCloud(const PointCloud& point_cloud)
   CHECK_EQ(num_blocks, 0);
 }
 
-CompressedPointCloud::CompressedPointCloud(const std::vector<int32>& point_data,
-                                           size_t num_points)
-    : point_data_(point_data), num_points_(num_points) {}
+CompressedPointCloud::CompressedPointCloud(
+    const proto::CompressedPointCloud& proto) {
+  CHECK(proto.has_num_points()) << "The compressed point cloud is malformed.";
```

Comment:
This is not needed. It is quite common to interpret missing fields in a proto as their default value, i.e. no points in this case.